### PR TITLE
[GraphBolt] set buffer_size as whole itemset in default

### PR DESCRIPTION
**Pull request by Rhett-Ying**
## Description

In principle, we should shuffle upon the whole item set to achieve good randomness.

For now, accuracy aligns with https://github.com/dmlc/dgl/blob/master/examples/sampling/graphbolt/rgcn/README.md#accuracies

**use_indexing=True, buffer_size=-1**
Training~Epoch 02: 615it [01:01,  9.92it/s]
Evaluating the model on the training set.
Inference: 154it [00:17,  8.61it/s]
Finish evaluating on training set.
Evaluating the model on the validation set.
Inference: 16it [00:01,  9.47it/s]
Finish evaluating on validation set.
Evaluating the model on the test set.
Inference: 11it [00:01,  8.93it/s]
Finish evaluating on test set.
Run: 01, Epoch: 03, Loss: 1.8374, Train: 65.51%, Valid: 41.73%, Test: 41.36%
Finish evaluating on test set.
Run 01:
Highest Train: 65.51
Highest Valid: 42.13
  Final Train: 59.28
   Final Test: 41.40

**use_indexing=False, BF=-1**
Run: 01, Epoch: 02, Loss: 2.0556, Train: 60.57%, Valid: 42.76%, Test: 42.09%
Finish evaluating on test set.
Training~Epoch 02: 615it [01:51,  5.51it/s]
Evaluating the model on the training set.
Inference: 154it [00:29,  5.19it/s]
Finish evaluating on training set.
Evaluating the model on the validation set.
Inference: 16it [00:03,  5.33it/s]
Finish evaluating on validation set.
Evaluating the model on the test set.
Inference: 11it [00:02,  5.12it/s]
Finish evaluating on test set.
Run: 01, Epoch: 03, Loss: 1.7843, Train: 66.54%, Valid: 42.68%, Test: 41.60%
Finish evaluating on test set.
Run 01:
Highest Train: 66.54
Highest Valid: 42.76
  Final Train: 60.57
   Final Test: 42.09

## Checklist
Please feel free to remove inapplicable items for your PR.
- [ ] The PR title starts with [$CATEGORY] (such as [NN], [Model], [Doc], [Feature]])
- [ ] I've leverage the [tools](https://docs.google.com/document/d/1iHyj7zlmygKSk5gBPsqIqL5ASPzJSPREaNT_QdsiYA4/edit) to beautify the python and c++ code.
- [ ] The PR is complete and small, read the [Google eng practice (CL equals to PR)](https://google.github.io/eng-practices/review/developer/small-cls.html) to understand more about small PR. In DGL, we consider PRs with less than 200 lines of core code change are small (example, test and documentation could be exempted).
- [ ] All changes have test coverage
- [ ] Code is well-documented
- [ ] To the best of my knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change
- [ ] Related issue is referred in this PR
- [ ] If the PR is for a new model/paper, I've updated the example index [here](../examples/README.md).

### CODE REVIEW

**Comment by Skeleton003** [comment]
> ## Description
> In principle, we should shuffle upon the whole item set to achieve good randomness.
> 
> For now, accuracy aligns with https://github.com/dmlc/dgl/blob/master/examples/sampling/graphbolt/rgcn/README.md#accuracies
> 
> **use_indexing=True, buffer_size=-1** Training~Epoch 02: 615it [01:01, 9.92it/s] Evaluating the model on the training set. Inference: 154it [00:17, 8.61it/s] Finish evaluating on training set. Evaluating the model on the validation set. Inference: 16it [00:01, 9.47it/s] Finish evaluating on validation set. Evaluating the model on the test set. Inference: 11it [00:01, 8.93it/s] Finish evaluating on test set. Run: 01, Epoch: 03, Loss: 1.8374, Train: 65.51%, Valid: 41.73%, Test: 41.36% Finish evaluating on test set. Run 01: Highest Train: 65.51 Highest Valid: 42.13 Final Train: 59.28 Final Test: 41.40
> 
> **use_indexing=False, BF=-1** Run: 01, Epoch: 02, Loss: 2.0556, Train: 60.57%, Valid: 42.76%, Test: 42.09% Finish evaluating on test set. Training~Epoch 02: 615it [01:51, 5.51it/s] Evaluating the model on the training set. Inference: 154it [00:29, 5.19it/s] Finish evaluating on training set. Evaluating the model on the validation set. Inference: 16it [00:03, 5.33it/s] Finish evaluating on validation set. Evaluating the model on the test set. Inference: 11it [00:02, 5.12it/s] Finish evaluating on test set. Run: 01, Epoch: 03, Loss: 1.7843, Train: 66.54%, Valid: 42.68%, Test: 41.60% Finish evaluating on test set. Run 01: Highest Train: 66.54 Highest Valid: 42.76 Final Train: 60.57 Final Test: 42.09
> 
> ## Checklist
> Please feel free to remove inapplicable items for your PR.
> 
> * [ ]  The PR title starts with [$CATEGORY] (such as [NN], [Model], [Doc], [Feature]])
> * [ ]  I've leverage the [tools](https://docs.google.com/document/d/1iHyj7zlmygKSk5gBPsqIqL5ASPzJSPREaNT_QdsiYA4/edit) to beautify the python and c++ code.
> * [ ]  The PR is complete and small, read the [Google eng practice (CL equals to PR)](https://google.github.io/eng-practices/review/developer/small-cls.html) to understand more about small PR. In DGL, we consider PRs with less than 200 lines of core code change are small (example, test and documentation could be exempted).
> * [ ]  All changes have test coverage
> * [ ]  Code is well-documented
> * [ ]  To the best of my knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change
> * [ ]  Related issue is referred in this PR
> * [ ]  If the PR is for a new model/paper, I've updated the example index [here](../examples/README.md).
> 
> ## Changes

Did you test in fanout order [25, 10] or [10, 25]?

**Comment by Rhett-Ying** [comment]
@Skeleton003 Not tested with `fanout` as current implementation is same as DGL
